### DISCUSSION TITLE
Monitoring cluster

### DISCUSSION THREAD

**Discussion by Wjl-Cody** [apache/skywalking#10147] [Q&A]
Whether skywalking can monitor the information of all instances under the cluster in batches by deploying an agent once, or whether each service must install an agent

**Comment by wu-sheng** [comment]
If you are talking about Java agent thing, you need every service could access it from I/O(disk). That is a JVM requirement.

**Reply by Wjl-Cody** [reply]
Whether the skywalking java agent can be deployed to each machine in the cluster in batch, similar to one click deployment to all machines in the cluster, instead of each machine having to be deployed separately

**Reply by wu-sheng** [reply]
That is your own responsibility. SkyWalking doesn't limit or require how you deploy the system. 
What you asked could be easily done by any mature CD system.
